### PR TITLE
[TASK] Update the contribution workflow documentation

### PR DESCRIPTION
**Pull request by oliverklee**
We're not using Travis CI anymore. Also, there is no way external contributors to request reviews on pull requests.

Also update outdated URLs.

Also update the wording to be less Emogrifier-specific so we can copy'n'paste this documentation more easily to our sister project.

### CODE REVIEW

File: .github/CONTRIBUTING.md
Comment (JakeQZ) [inline .github/CONTRIBUTING.md:33]: First-time contributors will need to wait for an admin to allow the CI build to run.  I think it's worth noting such here.

File: .github/CONTRIBUTING.md
Comment (JakeQZ) [inline .github/CONTRIBUTING.md:22]: I know this doesn't relate specifically to this PR, but...

I think both 'remote' and 'upstream' should be in inverted-ticks, and I'm not sure 'local' belongs.  'A local remote 
"upstream"' does not appear to make any sense.  Though I don't fully understand what's going on here, and not sure I ever did that step.

The link given provides a link in turn for adding a `remote` `upstream", so an extra link for that probably isn't needed (though might be considered).

**Review by JakeQZ — CHANGES_REQUESTED** [review]
Looks mostly good, though Git should be capitalized, and there are a couple of other points to consider.

File: .github/CONTRIBUTING.md
Comment (oliverklee) [reply .github/CONTRIBUTING.md:22]: I think I'll just remove that step. GitHub has added some UI for that anyway in the meantime.

File: .github/CONTRIBUTING.md
Comment (JakeQZ) [inline .github/CONTRIBUTING.md:33]: Sorry to be nitpicky, but this implies that a maintainer would have approved the changes, whereas all they've done is clicked a button to allow the CI build to run.

```suggestion
   maintainer to allow your CI build to run.
```

**Review by JakeQZ — CHANGES_REQUESTED** [review]
Sorry, I may be being nitpicky, but I think necessarily so ;)

Also, given this is markdown, rather then renumber the list, you could change all the item indexes to `1.`.

**Comment by oliverklee** [comment]
> Sorry, I may be being nitpicky, but I think necessarily so ;)

Honestly, I really appreciate your thorough reviews! (And me getting annoyed occasionally and then is a price I gladly pay for our code and documentation being so high-quality. :heart:)

File: .github/CONTRIBUTING.md
Comment (oliverklee) [reply .github/CONTRIBUTING.md]: Done.

**Review by JakeQZ — APPROVED** [review]
LGTM now.

**Pull request merged**
